tests/tests_pytorch/models/test_onnx.py

```diff
@@ -42,6 +43,20 @@ def test_model_saves_with_input_sample(tmp_path):
     assert os.path.getsize(file_path) > 4e2
 
 
+@RunIf(onnx=True)
+def test_model_saves_with_pathlib_file_path(tmp_path):
+    """Test that ONNX model saves with pathlib.Path file_path and size is greater than 3 MB."""
+    model = BoringModel()
+    trainer = Trainer(fast_dev_run=True)
+    trainer.fit(model)
```

Comment:
Pretty sure it's not necessary to train. We can export the model directly.

Reply:
```suggestion
```

Reply:
Good spot - I've updated the test to combine other `os.path` and `pathlib.Path` cases (and removed the `.fit(...)` call at the same time): https://github.com/Lightning-AI/pytorch-lightning/pull/19727/commits/340c9fa01aa93f88f90008dff0b40384adb1bb83